Issue#190 Create Project Page

BaaniLeen opened the pull request:
### Description

Added a Project page to the current code base, which takes the information from an array and loads in it separate cards. Ensured high modularity & no redundancy of code.
Also updated the README file, as discussed with @ancyp .

Fixes #190 

Time Taken : 13 hours
6 hours (Making the modular Projects cards) + 3 hours (Tabs and pills) + 4 hours(Accomodated feedback from mentors - Added modal view; Ensured Mobile responsiveness of buttons on cards; Tried changing Title of the app)

### Type of Change:
**Delete irrelevant options.**

- Code
- Quality Assurance
- User Interface
- Outreach
- Documentation
  

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

### Mockups
Demo:
https://drive.google.com/file/d/1QRG3f7QhJU2NZfs-KuIfHAE2TDWcBAoG/view?usp=sharing

### Mobile Responsiveness : Tested.

### Checklist:
**Delete irrelevant options.**

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
  
**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

ancyp commented:
@BaaniLeen acc to timeline I see
Modal window(Fig 2) : Summary of Aim, Technical Stack used, Project specific resources, Slack channel name.
Project Maintainers for specific projects.(Handles for them)

Was this decision changed ?

BaaniLeen commented:
@ancyp Yes, initially the idea was to have information about the contributors or maintainers but to keep it dynamic May had suggested to use handles in the last All teams meeting, which she can update like @systers/maintainers-gh-pages for github pages, etc. So since we have a small amount of data which can be easily represented on a card, I did not implement the modal view. Let me know your views about it. :)

ancyp commented:
Can you update desc to add if this is tested on mobile for this and all future PRs?
Are you creating a new issue for navbar responsiveness?

BaaniLeen commented:
Sure @ancyp I will add that. 
As for the navbar, I will make a new issue to test the mobile responsivenss of the current version of the app till now :)

ancyp commented:
Some design choices:
@Janiceilene I'm not sure if image is needed. 
Summary, Project specific resources, Slack channel name instead makes more sense.
I would also prefer the repo button being replaced by hyper links on the name of the project - saves space.
As there are a lot of projects, it would be nice to be a tabbed pane on top to show systers/peacecorp projects.

Janiceilene commented:
I'm still unsure of what the images are going to be? Was that decided? I think the github team name was being used instead of the Slack handle for folks who aren't on Slack, but I'm okay with either option.  I like the idea of hyperlinks instead of repo buttons and tabbed panes to move between Systers and PC projects.

divyanshu-rawat commented:
@BaaniLeen yes, I would also prefer the repo button being replaced with hyper links.

divyanshu-rawat commented:
@BaaniLeen kindly look into this as soon as screen size is reduced and collapsed navbar icon is clicked to expand the navbar , Navbar end up hiding the tet HOME PAGE.

Before 
![screen shot 2018-06-03 at 10 22 53 pm](https://user-images.githubusercontent.com/13464678/40888991-0bdd0a3e-677d-11e8-8225-88177f032883.png)

After
![screen shot 2018-06-03 at 10 22 58 pm](https://user-images.githubusercontent.com/13464678/40888993-16efd456-677d-11e8-9983-c90c1f3dee78.png)

divyanshu-rawat commented:
And when screen size is reduced further rest of th eli items in dropdown hide themself.
![screen shot 2018-06-03 at 10 26 28 pm](https://user-images.githubusercontent.com/13464678/40889006-4300d702-677d-11e8-8106-9cb1de617074.png)

BaaniLeen commented:
@ancyp @divyanshu-rawat @Janiceilene I have added mobile responsiveness, removed the images for mobile view and updated the repo button to the github icon. Let me know if this is okay or if I need to incorporate more suggestions.

BaaniLeen commented:
@ancyp @divyanshu-rawat @Janiceilene I have added mobile responsiveness, removed the images for mobile view and updated the repo button to the github icon. Let me know if there are any more suggestions. :)

BaaniLeen commented:
@divyanshu-rawat Is this tablet size? It works fine now, I guess.
![screen shot 2018-06-07 at 12 59 49 pm](https://user-images.githubusercontent.com/22029744/41084803-ba26ff50-6a52-11e8-88a2-4b74f5c56364.png)
Please let me know. :)

ancyp commented:
![image](https://user-images.githubusercontent.com/7522358/41135893-865eebac-6a88-11e8-8c9c-725396449f05.png)

The github icon should be black in color right?

The github icons needs to be aligned in every card.

IMO making the entire card clickable would be better than restricting the clickable area to the  icon. If there are multiple icons, it makes sense to restrict it to that specific icon. 

Did you have any better ideas than having a tabbed pane as Janice and I suggested?

Why aren't we adding slack information here?

ancyp commented on src/app/projects/projects.component.html:
```diff
@@ -1 +1,69 @@
+		<!-- Peace Corps Projects Note-->
+
+		<!-- Projects -->
+		<div class=" container box logos mx-auto d-block">
```
Extra space

BaaniLeen replied:
Sure. Will change that!
Thanks :)

ancyp commented on src/app/projects/projects.component.html:
```diff
@@ -1 +1,69 @@
+
+		<!-- Projects -->
+
+		<div class=" container box logos mx-auto d-block">
```
extra space

BaaniLeen replied:
Sure. Will change that!
Thanks :)

BaaniLeen commented:
@ancyp By the alignment of the Github icons do you mean, cards should be of the same sizes too?
I will change the colour of the Github icons to black. :)
Shall I switch to a tabbed pane for peace Corps and Systers projects? I felt the problem with this would be that the user would not be aware which of those categories of projects, he/she wants to look for. What are your views about it?
Regarding the slack icon, since the github icon too is at the bottom, I can incorporate both of them in a single line itself. Regarding making the entire card clickable, that is actually a great idea! :)
But since we are adding the slack symbol, so making the entire card clickable here would not make much sense.
Or maybe, I can just have the description and the project name on the cards, make it clickable and a modal opens up, with details including Github symbol and slack symbol - which should redirect to the project specific channel.
Which one suits better? @ancyp @Janiceilene

Janiceilene commented:
If we add a slack icon, what would it actually link to? If it's just to join the slack org, it wouldn't be super useful. Can it link to the specific Slack channel and redirect to the join page if they're not already signed up for our organization? If this is possible, then adding a Slack icon in line with the Github icon would be great. At that point you wouldn't want the whole card to be clickable.

Also, we should add a sentence explanation of what each project is, though that might make each card too long.

I think tabbed panes for Systers and Peace Corps projects, like the ones Anumeha has for the different communities, would be nice. That way people can still browse through them, but there aren't as many to sift through if you're looking for something in particular. I don't like filtering when someone wants to browse, that's too limiting, but tabs are nice.

I'm also okay with the name and description being on the card, then a modal that has the name, description, technologies, github, and name of the Slack channel.

I don't think there was ever a decision made as to what all of the images will actually be for each repo. I'm still worried that there won't be images for most of these repos.

BaaniLeen commented:
@ancyp @divyanshu-rawat @Janiceilene Here is the new demo for the Project Page with the modals and Pills.
The two sub - issues I am unable to address right now are : 
1) Align all the view buttons at the same level. Since the descripiton of each project varies, so it is difficult to match the position of the View More buttons at the same level. Please help me find a solution to this, if this is still needed.
2) I am unable to rename the title of the app. I have tried following alot of resources, the most reliable one being https://angular.io/guide/set-document-title, but it still does not work. Can you please help me with these.
If these 2 issues are not directly related, can you please approve and merge the PRs? :)
Thanks. Let me know if there are any more suggestions. :)

ancyp commented:
changing in this file https://github.com/systers/systers.github.io/blob/gsoc18-code/src/index.html does not work?

There is a property called 
 min-height: 10;
  max-height: 10;
for panels,
 https://stackoverflow.com/questions/24450010/how-to-make-bootstrap-panel-body-with-fixed-height does this work?

BaaniLeen commented:
@ancyp Thank you! Have updated the title name.
For the constant height cards, it works, but I am still not able to same alignment of footers. I tried using Bootstrap card-footer class. Somehow the alignment of the footer is coming beyond the card, and I am trying to find a solution to this before I commit this. 

Here is how it looks : 
![screen shot 2018-06-16 at 12 08 44 pm](https://user-images.githubusercontent.com/22029744/41496508-11f5adb8-715f-11e8-8dc3-160a77bb7220.png)

Can we make this into a separate issue and merge this PR?

BaaniLeen commented:
@ancyp Can you please review and merge this PR? :)

ancyp commented on src/app/projects/projects.component.html:
```diff
@@ -1 +1,125 @@
+									    aria-hidden="true">
+										<div class="modal-dialog modal-notify modal-success" role="document">
+											<div class="modal-content">
+												<div class="modal-header" style="background-color: #8d2165;">
```
move to css

BaaniLeen replied:
@ancyp Same problem as in the other PR.
I have shifted the style attribute from the paragraph tag to the scss file, but for the model-header class, the style property for background colour when put in the scss file, is overridden by the background-color property (green) of model-header class in the mdBootstrap stylesheet file, since perhaps the newcomers.component.scss file is accessed before the the mdBootstrap default css file and thus the green colour replaces the maroon one mentioned in the css file.
In the case that I add the same statement in the style attribute, it overrides all the existing scss imported attributes and thus the modal-header appears maroon. So in the case we want to remove the style attribute of the modal-header, one would need to go into the depth and analyse how Angular is loading its files. So can we do with the current commit or can we create a separate issue to solve this one, later?

ancyp commented:
@divyanshu-rawat @Janiceilene @Tharangi @kanmodi can we get one more mentor's approval/review please?

kanmodi commented:
Looks good to me!

BaaniLeen commented:
@ancyp @Janiceilene Ready to merge :)

Tharangi commented:
@ancyp I will merge this PR
